pkg/webhook/validator.go

```diff
@@ -574,9 +620,39 @@ func ValidatePolicyAttestationsForAuthority(ctx context.Context, ref name.Refere
 	return ret, nil
 }
 
+// ResolvePodScalable implements policyduckv1beta1.PodScalableValidator
+func (v *Validator) ResolvePodScalable(ctx context.Context, ps *policyduckv1beta1.PodScalable) {
+	if apis.IsInDelete(ctx) {
+		// Don't mess with things that are being deleted.
+		return
+	}
+
+	if apis.IsInUpdate(ctx) {
+		newReplicaCount := ps.Spec.Replicas
+		original := apis.GetBaseline(ctx).(*policyduckv1beta1.PodScalable)
+		if newReplicaCount != nil && original != nil && original.Spec.Replicas != nil {
+			if *newReplicaCount < *original.Spec.Replicas {
+				logging.FromContext(ctx).Debugf("Skipping validations due to scale down request %s/%s", &ps.ObjectMeta.Name, &ps.ObjectMeta.Namespace)
+				return
+			}
+		}
+	}
+
+	imagePullSecrets := make([]string, 0, len(ps.Spec.Template.Spec.ImagePullSecrets))
+	for _, s := range ps.Spec.Template.Spec.ImagePullSecrets {
+		imagePullSecrets = append(imagePullSecrets, s.Name)
+	}
+	opt := k8schain.Options{
+		Namespace:          ps.Namespace,
+		ServiceAccountName: ps.Spec.Template.Spec.ServiceAccountName,
+		ImagePullSecrets:   imagePullSecrets,
+	}
+	v.resolvePodSpec(ctx, &ps.Spec.Template.Spec, opt)
+}
+
 // ResolvePodSpecable implements duckv1.PodSpecValidator
 func (v *Validator) ResolvePodSpecable(ctx context.Context, wp *duckv1.WithPod) {
-	if wp.DeletionTimestamp != nil {
+	if apis.IsInDelete(ctx) {
```

Comment:
You should check that `IsInDelete` checks the deletion timestamp, since we don't want to mess with things being finalized either.